packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -94,13 +168,114 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
+///  * [FlutterGoldensRepositoryFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images through Skia Gold.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  FlutterSkiaGoldFileComparator(Uri basedir) : super(basedir) {
+    _skiaClient = SkiaGoldClient(fs.directory(basedir));
+  }
+
+  /// Constructor used specifically for testing in order to provide mock members.
+  FlutterSkiaGoldFileComparator.test(
+    Uri basedir,
+    FileSystem testFileSystem,
+    Platform testPlatform,
+  ) : super(
+    basedir,
+    fs: testFileSystem,
+    platform: testPlatform,
+  );
+
+
+  SkiaGoldClient _skiaClient;
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    GoldensClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= GoldensClient();
+
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(goldens, defaultComparator);
+    final SkiaGoldClient skiaClient = SkiaGoldClient(baseDirectory);
+    await skiaClient.auth();
+    await skiaClient.imgtestInit();
+    return FlutterSkiaGoldFileComparator(baseDirectory.uri);
+  }
+
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
+
+    final File goldenFile = getGoldenFile(golden);
+    if (!goldenFile.existsSync()) {
+      throw TestFailure('Could not be compared against non-existent file: "$golden"');
+    }
+    return await _skiaClient.imgtestAdd(golden.path, goldenFile);
+  }
+
+  /// Prepends the golden Uri with the library name that encloses the current
+  /// test.
+  Uri _addPrefix(Uri golden) {
+    final String prefix = basedir.pathSegments[basedir.pathSegments.length - 2];
+    return Uri.parse(prefix + '.' + golden.toString());
+  }
+}
+
+class FlutterDummyGoldenFileComparator extends FlutterGoldenFileComparator {
+  FlutterDummyGoldenFileComparator(Uri basedir) : super(basedir);
+
+  /// Creates a new [FlutterDummyGoldenFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    return FlutterSkiaGoldFileComparator(defaultComparator.basedir);
+  }
+
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    print('Skipping "$golden" test : Skia Gold is not available in this testing'
+      ' environment and flutter/goldens repository comparison is only available'
+      ' on Linux machines.');
```

Comment:
nit: the closing `)` should have the same indentation as the line with the opening `(`

Reply:
Also, move the spaces to the end of a line, not the beginning of the next.

I'd probably do:
```dart
    print(
      'Skipping "$golden" test : Skia Gold is not available in this testing '
      'environment and flutter/goldens repository comparison is only available '
      'on Linux machines.'
    );
```